### PR TITLE
Getting to grips with dokka upgrade

### PR DESCRIPTION
**Pull request by yschimke**
Expectation is being able to run `./gradlew dokkaGfmMultiModule` successfully with results in same place.

https://github.com/Kotlin/dokka/blob/master/runners/gradle-plugin/MIGRATION.md

https://kotlin.github.io/dokka/1.4.10.2/user_guide/gradle/usage/

### CODE REVIEW

**Comment by yschimke** [comment]
Failing with 

```
Caused by: java.lang.IllegalArgumentException: Count 'n' must be non-negative, but was -1.
```

File: build.gradle [hunk 1]
```diff
@@ -243,24 +246,42 @@ subprojects { project ->
...
+          suppress = true
+        }
+        externalDocumentationLink {
+          url.set(URL("https://square.github.io/okio/2.x/okio/"))
+          packageListUrl.set(URL("https://square.github.io/okio/2.x/okio/package-list"))
+        }
       }
     }
   }
+//  dokka {
```
Comment (swankjesse) [inline build.gradle:267]: remove?

**Review by swankjesse — APPROVED** [review]

**Comment by yschimke** [comment]
From master (before PR)

image

**Comment by yschimke** [comment]
Reported current error https://github.com/Kotlin/dokka/issues/1592

**Comment by yschimke** [comment]
TODO - next steps build locally on this PR temporarily to see what published docs would be like.

**Comment by yschimke** [comment]
Parking for now...

A bug upstream is reported.
There are some broken links but site is browseable
Too many modules are generated currently.
Results are terrible currently, lots of [jvm] in the visible output, presumable hooks for the javascript or similar?

File: okhttp/src/main/kotlin/okhttp3/HttpUrl.kt [hunk 2]
```diff
@@ -424,7 +423,7 @@ class HttpUrl internal constructor(
    * The username, or an empty string if none is set.
    *
    * | URL                              | `encodedUsername()` |
-   * | :------------------------------- | :------------------ |
```
Comment (yschimke) [inline okhttp/src/main/kotlin/okhttp3/HttpUrl.kt:427]: This works around the upstream bug.

**Pull request closed without merging**